azerothcore/azerothcore-wotlk
# fix(Core/Misc): all GCC warnings

## CHANGES PROPOSED:
-  Fixed most of GCC warnings
-  Disabled a couple of GCC warnings in order to have a clean build, which will allow adding GCC to our CI. The goal is merely to prevent more warnings to get to master. I will open issues for those ignored warnings so we can fix them later.


## TESTS PERFORMED:
- None


## HOW TO TEST THE CHANGES:

Theoretically, this PR should not introduce any change in the game, however it's always better to double check, so please:

- checkout this PR and play the game for a while, make sure there is nothing broken by this change

I know it's too generic, but I can't provide more detailed test instructions this time.



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
